hal_st/middlewares/ble_middleware/GattClientSt.cpp

```diff
@@ -143,6 +209,11 @@ namespace hal
 
         if (onDiscoveryCompletion)
```

Comment:
onDiscoveryCompletion and onDone, can they both be in progress at the same time?
If not, we do not need to try to execute both.
And, how do we check onDiscoveryCompletion against nullptr like we do for onDone?

Reply:
No, they can't be in progress at same time. onDiscoveryCompletion is used only during the discovery process, as soon as has been finished, it will be nullptr.

We just try to execute that one that is not nullptr, are you suggesting to add an "else" condition for onDiscoveryCompletion and put onDone call there?

We are already checking if onDiscoveryCompletion is nullptr, line 210.

Reply:
What I meant by streamlining is something like this:
```c++
        if (onDiscoveryCompletion)
            infra::Subject<services::GattClientDiscoveryObserver>::NotifyObservers(std::exchange(onDiscoveryCompletion, nullptr));
        else if(onDone)
            std::exchange(onDone, nullptr)();
        
        
```

Reply:
Oh, I see now, thanks. And it is fixed.